lib/Serialization/Serialization.cpp

```diff
@@ -2715,8 +2715,10 @@ void Serializer::writeDecl(const Decl *D) {
                           nullptr, /*sorted=*/true);
 
     SmallVector<TypeID, 8> inheritedAndDependencyTypes;
-    for (auto inherited : extension->getInherited())
+    for (auto inherited : extension->getInherited()) {
+      assert(!inherited.getType()->hasArchetype());
```

Comment:
It would be nice if in the future, we could avoid serializing archetypes outside of SIL completely

Reply:
Agreed! Chipping away at it slowly.